src/sentry/dynamic_sampling/sliding_window_rebalancing.py

```diff
@@ -0,0 +1,123 @@
+import logging
+import time
+from collections import defaultdict
+from datetime import datetime, timedelta
+from typing import List, Mapping, Optional, Sequence, Tuple
+
+from snuba_sdk import (
+    Column,
+    Condition,
+    Direction,
+    Entity,
+    Function,
+    Granularity,
+    LimitBy,
+    Op,
+    OrderBy,
+    Query,
+    Request,
+)
+
+from sentry import options
+from sentry.dynamic_sampling.rules.utils import OrganizationId, ProjectId
+from sentry.dynamic_sampling.snuba_utils import MAX_TRANSACTIONS_PER_PROJECT
+from sentry.sentry_metrics import indexer
+from sentry.snuba.dataset import Dataset, EntityKey
+from sentry.snuba.metrics.naming_layer.mri import TransactionMRI
+from sentry.snuba.referrer import Referrer
+from sentry.utils.snuba import raw_snql_query
+
+logger = logging.getLogger(__name__)
+MAX_SECONDS = 60
+CHUNK_SIZE = 9998  # Snuba's limit is 10000, and we fetch CHUNK_SIZE+1
+QUERY_TIME_INTERVAL = timedelta(
+    hours=24
+)  # By default, we want to get the volume of the last 24 hours.
+
+
+def fetch_projects_with_total_root_transactions_count(
+    org_ids: List[int],
+    granularity: Optional[Granularity] = None,
+    query_interval: Optional[timedelta] = None,
+) -> Mapping[OrganizationId, Sequence[Tuple[ProjectId, int]]]:
```

Comment:
This functionality is fully contained in `fetch_projects_with_total_volumes` no need to re-write the same query.

`fetch_projects_with_total_volumes` in addition gets `keep` and `drop` counts which it gets wrongly... but you don't need them here.

After my fix of the rebalance bias we shouldn't need `fetch_projects_with_total_volumes` to return `drop` and `keep` so one of the two functions should go (i.e. either keep yours and use it for the project rebalance bias or the other way).

(My fault for not removing `keep` and `drop` from `fetch_projects_with_total_volumes`).

Reply:
I would prefer to refactor in a followup PR, since I don't want to touch any code related to prioritise projects. I will add a TODO and a ticket.